File: app/src/main/java/com/thoughtworks/pafsilva/testautomationworkshop/login/abstractions/LoginView.java
package com.thoughtworks.pafsilva.testautomationworkshop.login.abstractions;

import com.thoughtworks.pafsilva.testautomationworkshop.model.User;

public interface LoginView {
    void showLoading();
    void dismissLoading();
    void showErrorDialog(String errorMessage);
    void doLogin();
    void navigateToBrowser();
    void setEmailEditTextError(String errorMessage);
    void setPasswordEditTextError(String errorMessage);
    void navigateToUserDetails(User user);
}

File: app/src/main/java/com/thoughtworks/pafsilva/testautomationworkshop/login/abstractions/LoginView.kt
package com.thoughtworks.pafsilva.testautomationworkshop.login.abstractions

import com.thoughtworks.pafsilva.testautomationworkshop.model.User

interface LoginView {
    fun showLoading()
    fun dismissLoading()
    fun showErrorDialog(errorMessage: String)
    fun doLogin()
    fun navigateToBrowser()
    fun setEmailEditTextError(errorMessage: String)
    fun setPasswordEditTextError(errorMessage: String)
    fun navigateToUserDetails(user: User)
}
